experiment/nipopow_verifier/tools/proof/proof.py

```diff
@@ -75,3 +89,175 @@ def set(self, proof, proof_name=""):
             self.hashed_headers.append(Hash(self.proof[i][0]))
 
         self.size = len(self.proof)
+
+        (
+            self.best_level,
+            self.best_score,
+            self.levels,
+            self.scores,
+        ) = best_level_and_score(self.proof)
+
+        for level in sorted(self.levels.keys()):
+            print(
+                "Level", level, "has", self.levels[level], "blocks with score", end=""
+            )
+            if level == self.best_level:
+                print(" " + str(self.best_score) + " <- best")
+            else:
+                print(" " + str(self.scores[level]))
+
+        if header is None or header_map is None or interlink_map is None:
+            print("No interlink provided")
+            return
```

Comment:
This can be an assert with the specified message

Reply:
It is not necessarily an error. A proof can exist without having including best sub-proof inside.